dags/gmail_data_import_pipeline.py

```diff
@@ -0,0 +1,170 @@
+# Note: DagBag.process_file skips files without "airflow" or "DAG" in them
+
+import os
+import logging
+from datetime import timedelta
+from tempfile import TemporaryDirectory
+from pathlib import Path
+
+from google.cloud import bigquery
+
+from data_pipeline.utils.data_store.bq_data_service import (
+    create_table_if_not_exist,
+    load_file_into_bq,
+)
+
+from data_pipeline.utils.data_store.s3_data_service import (
+    download_s3_json_object,
+)
+
+from data_pipeline.utils.dags.data_pipeline_dag_utils import (
+    create_dag,
+    create_python_task
+)
+
+from data_pipeline.gmail_data.get_gmail_data_config import (
+    GmailGetDataConfig
+)
+
+from data_pipeline.utils.pipeline_file_io import (
+    get_yaml_file_as_dict
+)
+
+from data_pipeline.gmail_data.get_gmail_data import (
+    get_gmail_service_for_user_id,
+    get_label_list,
+    write_dataframe_to_csv_file
+)
+
+GMAIL_SCOPES = ['https://www.googleapis.com/auth/gmail.readonly']
+GMAIL_ACCOUNT_SECRET_FILE = 'DELETE/elife-data-pipeline-test.json'
+
+GMAIL_DATA_USER_ID_ENV = "GMAIL_DATA_USER_ID"
+GMAIL_DATA_CONFIG_FILE_PATH_ENV_NAME = "GMAIL_DATA_CONFIG_FILE_PATH"
+DEPLOYMENT_ENV_ENV_NAME = "DEPLOYMENT_ENV"
+DEFAULT_DEPLOYMENT_ENV = "ci"
+
+DAG_ID = "Gmail_Data_Pipeline"
```

Comment:
minor: thank you for changing. Just noticed still not completely matching the filename, with the current filename it would be `Gmail_Data_Import_Pipeline`.

Reply:
I think the file is for importing the job to pipeline but the dag name is pipeline itself ?

Reply:
I would agree to some degree. But I would argue that the purpose of the file is to define the DAG (as it is in the `dags` directory), as such it seems to make sense for them to be consistent. (I think I may have read something about best practices but can't find it now)

The other argument would be that it would be consistent with what we do in other DAGs, I think.

Whether we should have `pipeline` in the filename and DAG ID is another question. We have for example controller DAGs that don't.

Reply:
OK I see, I am updating